src/psql_erd_loader.rs

```diff
@@ -102,11 +102,18 @@ pub struct PostgreSqlERDLoader {
 }
 
 impl PostgreSqlERDLoader {
-    pub fn new(
+    pub async fn new(
         connection_string: &str,
         schema_name: String,
     ) -> Result<PostgreSqlERDLoader, SqlantError> {
-        let client = Client::connect(connection_string, NoTls)?;
+        let (client, connection) = tokio_postgres::connect(connection_string, NoTls).await?;
+        // The connection object performs the actual communication with the database,
+        // so spawn it off to run on its own.
+        tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                eprintln!("connection error: {}", e);
+            }
+        });
```

Comment:
I'll take a look at this and the above.
Could you look at the other two PRs, so I can rebase this one after the conflicts they will create

Reply:
If I attempt to put it into the struct, I get

```rust
error[E0382]: use of moved value: `connection`
   --> src/psql_erd_loader.rs:120:13
    |
110 |           let (client, connection) = tokio_postgres::connect(connection_string, NoTls).await?;
    |                        ---------- move occurs because `connection` has type `Connection<Socket, NoTlsStream>`, which does not implement the `Copy` trait
...
113 |           tokio::spawn(async move {
    |  ______________________-
114 | |             if let Err(e) = connection.await {
    | |                             ---------- variable moved due to use in coroutine
115 | |                 eprintln!("connection error: {}", e);
116 | |             }
117 | |         });
    | |_________- value moved here
...
120 |               connection,
    |               ^^^^^^^^^^ value used here after move

Some errors have detailed explanations: E0382, E0432.
```

looking for other alternatives ...

Reply:
I havent found a solution for either of these problems, except exposing `Connection` in the sqlant API, which IMO is worse.